src/FirmwarePlugin/APM/APMFirmwarePlugin.cc

```diff
@@ -954,9 +954,11 @@ void APMFirmwarePlugin::_handleRCChannels(Vehicle* vehicle, mavlink_message_t* m
         mavlink_rc_channels_t   channels;
 
         mavlink_msg_rc_channels_decode(message, &channels);
-        //-- Ardupilot uses 0-255 to indicate 0-100% where QGC expects 0-100
+        //-- Ardupilot uses 0-254 to indicate 0-100% where QGC expects 0-100
         if(channels.rssi) {
-            channels.rssi = static_cast<uint8_t>(static_cast<double>(channels.rssi) / 255.0 * 100.0);
+            if(channels.rssi != 255) { // As per mavlink specs, 255 means invalid, we must leave it like that for indicators to hide if no rssi data
+                channels.rssi = static_cast<uint8_t>(static_cast<double>(channels.rssi) / 255.0 * 100.0);
+            }
```

Comment:
There is no need to nested if checks, you can perform everything on the same if.
There is also a missing modification on the equation for 100%, where it should be `254.0`.
```suggestion
        //-- Ardupilot uses 0-254 to indicate 0-100% where QGC expects 0-100
        // As per mavlink specs, 255 means invalid, we must leave it like that for indicators to hide if no rssi data
        if(channels.rssi && channels.rssi != 255) {
            channels.rssi = static_cast<uint8_t>(static_cast<double>(channels.rssi) / 254.0 * 100.0);
            }
        }
```

Reply:
Sorry, you are totally right, I was working on a different matter and I did this very careless, sorry about it. I fixed and rebased it, should be good now. Thanks for the review!